lib/specifications/repositoryTypes/git.js

```diff
@@ -119,17 +118,26 @@ function update(projectData) {
   return git.Repository.openBare(projectData.localPath)
     .then(function (repository) {
 
+      var fetchOptions = {};
+      fetchOptions.downloadTags = 1;
+      fetchOptions.prune = 1;
+      fetchOptions.bare = 1;
+      fetchOptions.updateFetchhead = 1;
+      fetchOptions.remoteCallbacks = ignoreSSLErrors;
+      fetchOptions.fetchOpts = {
+        callbacks: ignoreSSLErrors
+      };
+
       // BUG: This isn't working at all, no new tags or branch are pulled down,
       // current heads aren't updated.
       // https://github.com/nodegit/nodegit/issues/742
-      return repository.fetch(repository.getRemote('origin'), {
-        callback: ignoreSSLErrors,
-        downloadTags: 1,
-        prune: 1
-      });
+      return repository.fetch('origin', fetchOptions);
     })
     .then(function () {
       return getProjectData(projectData, projectData.currentBranchName);
+    })
+    .catch(function (error) {
+      console.log("Error fetching from remote: ", error);
```

Comment:
We shouldn't catch the error here, we should make sure the promise chain is returned so that the catch at the rendering level catches it and reports it to screen like the other errors. That way end users can tell us exactly what went wrong. Later on when we want to hide the actual stack trace we can dump it into a mailto: link which will set up a reporting email for the user. If you really want to log this here then rethrow the error as well.

I think we already have http(s) access logs, if you want an error logging solution we can definitely put one in.